logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/message/LastReadContentHandler.kt

```diff
@@ -0,0 +1,39 @@
+package com.wire.kalium.logic.sync.receiver.message
+
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.id.IdMapper
+import com.wire.kalium.logic.data.message.Message
+import com.wire.kalium.logic.data.message.MessageContent
+import com.wire.kalium.logic.data.user.UserRepository
+import com.wire.kalium.logic.di.MapperProvider
+import com.wire.kalium.logic.util.TimeParser
+
+// This class handles the messages that arrive when some client has read the conversation.
+class LastReadContentHandler(
+    private val conversationRepository: ConversationRepository,
+    private val userRepository: UserRepository,
+    private val timeParser: TimeParser
+) {
+
+    // TODO: for now we are just handeling the case when the self user has read the conversation
+    // on another device, in the future we could handle the case when other user have read our message
+    // here too
+    suspend fun handle(
+        message: Message,
+        messageContent: MessageContent.LastRead
+    ) {
+        val isMessageComingFromOtherClient = message.senderUserId == userRepository.getSelfUserId()
+
+        if (isMessageComingFromOtherClient) {
```

Comment:
We should also only process the LastRead if it was sent in the `selfConversation`